Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -70,6 +73,29 @@ public override void Update(float frameTime)
                     continue;
 
                 comp.InfectedSecs += 1;
+                if (comp.InfectedSecs == -30)
+                {
+                    _popup.PopupEntity(Loc.GetString("zombie-infection-warning"), uid, uid);
+                }
+                else if (comp.InfectedSecs == 10 || comp.InfectedSecs == 30)
+                {
+                    _popup.PopupEntity(Loc.GetString("zombie-infection-underway"), uid, uid);
+                }
+
+                if (comp.InfectedSecs < 0)
+                {
+                    // This zombie has a latent virus, probably set up by ZombieRuleSystem
+                    continue;
```

Comment:
the flow here is weird to read and hardcoding the times is not a good idea, the warning should be moved to when you are zombified and underway stay but be declared like   `if (comp.InfectedSecs <= comp.UnderwayTimer (name subject to change))`